de.tu_bs.cs.isf.e4cf.core/src/de/tu_bs/cs/isf/e4cf/core/db/DatabaseFactory.java

```diff
@@ -0,0 +1,154 @@
+package de.tu_bs.cs.isf.e4cf.core.db;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+/**
+ * A singleton class to manage databases in the project.
+ * 
+ *
+ */
+public class DatabaseFactory {
+
+	private static DatabaseFactory INSTANCE = null;
+	private final String _JDBC_DRIVER = "org.sqlite.JDBC";
+
+	/**
+	 * Constructor
+	 * 
+	 */
+	private DatabaseFactory() {
+		// load JDBC-driver in driver-manager
+		try {
+			Class.forName(_JDBC_DRIVER);
+		} catch (ClassNotFoundException e) {
+			System.out.println(_JDBC_DRIVER + " can not be loaded.");
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * 
+	 * @return ConnectionFactory the instance of the class
+	 */
+	public synchronized static DatabaseFactory getInstance() {
+		if (null == INSTANCE) {
+			INSTANCE = new DatabaseFactory();
+		}
+		return INSTANCE;
+	}
+
+	/**
+	 * Method to create a database. The Database will be created if it does not
+	 * already exists.
+	 * 
+	 * @param pPath   String the path where to create the database
+	 * @param pDbName String the name of the database
+	 * @return Connection
+	 * @throws SQLException error while creating database
+	 */
+	public void createDatabase(final String pPath, final String pDbName) throws SQLException {
+		if (!databaseExists(pPath, pDbName)) {
+			final Connection con = DriverManager.getConnection("jdbc:sqlite:" + pPath + pDbName);

```

Comment:
Wir wissen nicht genau, wie DriverManager.getConnection funktioniert. Besteht die Möglichkeit, dass in Zeile 55 ein Fehler auftritt, die Datenbank nicht erstellt wird, und trotzdem Zeile 57 ausgegeben wird ?

Reply:
Fehler, die in Zeile 55 auftauchen können, wären: Entweder der SQLitetreiber nicht richtig geladen wurde (äußerst unwahrscheinlich) oder das Pfad (pPath) existiert nicht (also falsch eingegeben). In dem Fall stürzt das Programm ab und natürlich wird Zeile 57 nicht ausgegeben.